src/lib.rs

```diff
@@ -0,0 +1,280 @@
+use anyhow::{anyhow, bail, Ok, Result};
+use bb8_redis::{bb8::Pool, redis::cmd, RedisConnectionManager};
+use lazy_static::lazy_static;
+use std::{
+    fs::File,
+    net::IpAddr,
+    path::Path,
+    sync::{Arc, RwLock},
```

Comment:
@marioCluml I found it out a bit late, but this RwLock does not support asynchronous locking. It will cause unwanted blockings in async code blocks. Please switch to should `tokio::sync::RwLock`.

Reply:
Sounds good. I will make the adjustments and test it out